### PR TITLE
Fixing ASCII reader unable to read FD frames

### PR DESCRIPTION
**Pull request by sou1hacker**
CAN FD Frames in an ASCII log file could be logged in the following format:

```
0.045522 CANFD   62 Rx         7d  symbolic_name                    1 0 8  8 01 1b d6 d0 8a 1f d7 f6   104812  132   323000 a800f6c9 46500250 4b140250 20011736 2001040d
```

These frames are currently not read and skipped when using can.LogReader. This PR enables those frames to be read and processed.
Added unit tests for the same.

### CODE REVIEW

File: can/io/asc.py [hunk 1]
```diff
@@ -55,10 +55,14 @@ def __iter__(self):
             temp = line.strip()
             if not temp or not temp[0].isdigit():
                 continue
+            is_fd = False
             try:
                 timestamp, channel, dummy = temp.split(
                     None, 2
                 )  # , frameType, dlc, frameData
+                if channel == "CANFD":
+                    timestamp, _, channel, dir, dummy = temp.split(None, 4)
```
Comment (pierreluctg) [inline can/io/asc.py]: `dir` is a Python built-in name. Please use another variable name.

File: test/test_fd_vector.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        messages = []
+        for msg in reader:
+            messages.append(msg)
+
+        assert len(messages) == 4
+
+        first_message = messages[0]
+        assert first_message.timestamp == 0.045522
+        assert first_message.arbitration_id == 0x7D
+        assert first_message.is_extended_id is False
```
Comment (pierreluctg) [inline test/test_fd_vector.py]:
```suggestion
        assert not first_message.is_extended_id
```
or
```suggestion
        assert first_message.is_extended_id == False
```

File: test/test_fd_vector.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        assert first_message.channel == 61
+        assert first_message.dlc == 8
+        assert first_message.data == bytearray(
+            [0x01, 0x1B, 0xD6, 0xD0, 0x8A, 0x1F, 0xD7, 0xF6]
+        )
+        assert first_message.is_fd is True
```
Comment (pierreluctg) [inline test/test_fd_vector.py]:
```suggestion
        assert first_message.is_fd ==True
```

File: test/test_fd_vector.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        assert first_message.bitrate_switch is True
```
Comment (pierreluctg) [inline test/test_fd_vector.py]:
```suggestion
        assert first_message.bitrate_switch == True
```

File: test/test_fd_vector.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        assert first_message.error_state_indicator is False
```
Comment (pierreluctg) [inline test/test_fd_vector.py]:
```suggestion
        assert first_message.error_state_indicator == False
```

File: can/io/asc.py (hunk 1 above)
Comment (sou1hacker) [reply can/io/asc.py]: Done

**Comment by sou1hacker** [comment]
@pierreluctg Resolved all comments

**Comment by sou1hacker** [comment]
WIP: We are going to find equivalent ASCII files for those BLF files already present in test\data and add test cases for them.

**Comment by pierreluctg** [comment]
@hardbyte can you please review.

Thank you

**Comment by felixdivo** [comment]
@pierreluctg can this be merged?

**Comment by pierreluctg** [comment]
@felixdivo looks good to me.

**Pull request merged**